### PR TITLE
kube-scheduler: enable secure ports 10259

### PR DESCRIPTION
**Pull request by tanshanshan**
**What this PR does / why we need it**:
kube-scheduler: enable secure ports 10259

it depends on #64149
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by yue9944882** [comment]
@sttts bumping this thread. is this part of the series?

File: cmd/kube-scheduler/app/config/config.go [hunk 1]
```diff
@@ -63,12 +67,7 @@ type CompletedConfig struct {
 func (c *Config) Complete() CompletedConfig {
 	cc := completedConfig{c}
 
-	if c.InsecureServing != nil {
```
Comment (sttts) [inline cmd/kube-scheduler/app/config/config.go]: does these lines have to go?

File: cmd/kube-scheduler/app/options/options.go [hunk 2]
```diff
@@ -196,6 +211,7 @@ func (o *Options) Validate() []error {
 
 // Config return a scheduler config object
 func (o *Options) Config() (*schedulerappconfig.Config, error) {
+
```
Comment (sttts) [inline cmd/kube-scheduler/app/options/options.go]: nit: no new line here

File: cmd/kube-scheduler/app/testing/testserver.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+		s.SecureServing.Listener, s.SecureServing.BindPort, err = createListenerOnFreePort()
+		if err != nil {
+			return result, fmt.Errorf("failed to create listener: %v", err)
+		}
+		s.SecureServing.ServerCert.CertDirectory = result.TmpDir
+
+		t.Logf("kube-scheduler will listen securely on port %d...", s.SecureServing.BindPort)
+	}
+
+	if s.SecureServing.BindPort != 0 {
```
Comment (sttts) [inline cmd/kube-scheduler/app/testing/testserver.go]: this block is duplicated, isn't it?

File: test/integration/scheduler/serving_test.go [hunk 4]
```diff
@@ -0,0 +1,327 @@
+/*
```
Comment (sttts) [inline test/integration/scheduler/serving_test.go]: try to integrate into test/integration/controllermanager/serving_test.go, and rename that to test/integration/serving.

File: test/integration/serving/serving_test.go [hunk 5]
```diff
@@ -340,6 +355,277 @@ func testControllerManager(t *testing.T, tester controllerManagerTester, kubecon
 	}
 }
 
+func testSchedulerServing(t *testing.T) {
```
Comment (sttts) [inline test/integration/serving/serving_test.go]: Can't we rename testControllerManager to testComponent and integrate the scheduler test there as well?

**Comment by sttts** [comment]
What's the status of this (other than it needs a rebase)?

**Comment by sttts** [comment]
Completed the work in https://github.com/kubernetes/kubernetes/pull/69663. Thanks for doing the major part of the work!

**Pull request closed without merging**